### PR TITLE
feat(core): add cache diagnostics with stats and health check

### PR DESCRIPTION
**Pull request by laofahai**
## Summary
- `CacheManager.getStats()` — hits, misses, hit rate, evictions, per-namespace breakdown, memory estimates
- `InMemoryCacheProvider.keys()` — expose cache keys for diagnostics
- `createCacheHealthCheck()` — HealthCheckRegistry-compatible probe with optional hit rate threshold
- 21 tests covering stats tracking, health check pass/fail, edge cases

Partial progress on #73

## Test plan
- [x] 21 new tests pass locally
- [ ] CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added cache statistics (hits, misses, hit rate, evictions, estimated memory) with per-namespace breakdown and raw layer stats
  * Added cache health check with probe-based validation, configurable minimum hit-rate, and diagnostic metadata
  * Added ability to list cache keys

* **Tests**
  * Added comprehensive tests covering stats, namespaces, key listing, eviction/memory estimation, and health-check behaviors

### CODE REVIEW

**Comment by laofahai** [comment]
@coderabbitai review

File: packages/core/src/cache/cache-manager.ts [hunk 1]
```diff
@@ -148,6 +149,43 @@ export class CacheManager {
...
+   * Return comprehensive diagnostics including per-namespace breakdown
+   * and memory estimates. See spec §9.
+   */
+  getStats(options?: CacheManagerStatsOptions): CacheManagerStats {
+    const l1 = this.l1.stats();
+    const l2 = this.l2?.stats();
+
+    const totalHits = l1.hits + (l2?.hits ?? 0);
+    const totalMisses = l2?.misses ?? l1.misses;
+    const totalRequests = totalHits + totalMisses;
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/cache/cache-manager.ts:162]: Intentionally keeping `l2?.misses ?? l1.misses` — this represents CacheManager-level misses, not raw L1+L2 sums. When `get('key')` misses L1 and falls through to L2 which also misses, that's one CacheManager miss, not two. Summing L1+L2 would distort hitRate. Fixed the doc comment on `CacheManagerStats.misses` to clarify this is final-layer miss count.
